src/pykeen/nn/message_passing.py

```diff
@@ -61,7 +61,13 @@ def _reduce_relation_specific(
 
 
 class Decomposition(nn.Module, ABC):
-    """Base module for relation-specific message passing."""
+    """Base module for relation-specific message passing.
+
+    .. todo::
```

Comment:
@mberr @mali-git todo

Reply:
I think here we have forms of [matrix](https://en.wikipedia.org/wiki/Matrix_decomposition) / [tensor](https://en.wikipedia.org/wiki/Tensor_decomposition) decompositions or factorizations.

The authors search for a way to reduce the number of parameters needed by learning independent `d**2` matrices for each relation. The two proposed variants are treated as hyperparameters, and for different datasets different decompositions are superior in performance.

The decomposition module itself does not compute the full matrix from the factors, but rather provides efficient means to compute the product of the factorized matrix with the source nodes' latent features to construct the messages. This is usually more efficient than constructing the full matrices. 

For an intuition, you can think about a simple low-rank matrix factorization of rank 1, where `W = w w^T` for a `d`-dimensional vector `w`. Then, computing `Wv` as `(w w^T) v` gives you an intermediate result of size `d x d`, while you can also compute `w(w^Tv)`, where the intermediate result is just a scalar.